unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1542,20 +1542,21 @@ abstract public class BaseFPSAgentController {
             actionFinished(true);
         }
 
-        public virtual ObjectMetadata[] generateObjectMetadata() {
-            SimObjPhysics[] simObjects = null;
-            if (this.simObjFilter != null) {
-                simObjects = this.simObjFilter;
-            } else {
-                simObjects = GameObject.FindObjectsOfType<SimObjPhysics>();
+        public virtual ObjectMetadata[] generateObjectMetadata(SimObjPhysics[] simObjects = null) {
+            if (simObjects == null) {
+                if (this.simObjFilter != null) {
+                    simObjects = this.simObjFilter;
```

Comment:
This is a little surprising, passing null and getting metadata. Could be good for this function to just work with the simObjects it's passed, but I see the filtering needs to happen somewhere.

Reply:
`generateObjectMetadata` now has no default and will throw an error if passed a null value.